### PR TITLE
Updated docs to support resource classes in Score spec

### PR DESCRIPTION
**Pull request by sujaya-sys**
Added support for resource classes as described in https://github.com/score-spec/spec/issues/65

### CODE REVIEW

File: content/en/docs/reference/score-spec-reference.md [hunk 1]
```diff
@@ -91,7 +92,12 @@ resources:
     - **`Type`**: object.
     - **`Constraints`**: alphanumeric string key-value pairs.
 
-`resource-type`: specifies the resource in the target environment.
+`type`: specifies the resource type.
+
+- **Type**: string.
+- **Constraints**: alphanumeric string.
```
Comment (chrishumanitec) [inline content/en/docs/reference/score-spec-reference.md:98]: alphanumeric with dashes

File: content/en/docs/reference/score-spec-reference.md [hunk 2]
```diff
@@ -91,7 +92,12 @@ resources:
...
+
+`class`: a specialisation of the resource type. For example, a workload that needs an externally accessible resource might set the class to external while the workload that requires an encrypted resource might have a class of sensitive.
```
Comment (chrishumanitec) [inline content/en/docs/reference/score-spec-reference.md:100]: suggest replace `resource` with `storage bucket`

File: content/en/docs/reference/score-spec-reference.md (hunk 1 above)
Comment (chrishumanitec) [reply content/en/docs/reference/score-spec-reference.md:98]: Same for class below

**Review by chrishumanitec — CHANGES_REQUESTED** [review]

File: content/en/docs/reference/score-spec-reference.md (hunk 1 above)
Comment (sujaya-sys) [reply content/en/docs/reference/score-spec-reference.md:98]: I assume the same is true for `resource-name` and `annotations`? https://docs.score.dev/docs/reference/score-spec-reference/#resources

**Review by chrishumanitec — APPROVED** [review]
LGTM

**Pull request merged**